cli/doc/params.rs

```diff
@@ -1,87 +1,290 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+use super::display::{display_optional, SliceDisplayer};
+use super::ts_type::{ts_type_ann_to_def, TsTypeDef};
+use crate::swc_common::SourceMap;
 use crate::swc_ecma_ast;
+use crate::swc_ecma_ast::{ObjectPatProp, Pat, TsFnParam};
+use serde::Serialize;
+use std::fmt::{Display, Formatter, Result as FmtResult};
 
-use super::ts_type::ts_type_ann_to_def;
-use super::ParamDef;
-use super::ParamKind;
-use crate::swc_ecma_ast::Pat;
-use crate::swc_ecma_ast::TsFnParam;
+#[derive(Debug, Serialize, Clone)]
+#[serde(rename_all = "camelCase")]
+#[serde(tag = "kind")]
+pub enum ParamDef {
+  #[serde(rename_all = "camelCase")]
+  Array {
+    elements: Vec<Option<ParamDef>>,
+    optional: bool,
+    ts_type: Option<TsTypeDef>,
+  },
+  #[serde(rename_all = "camelCase")]
+  Assign {
+    left: Box<ParamDef>,
+    right: String,
```

Comment:
👍  great, @lucacasonato we'll have to update frontend to handle these changed structures